src/Workspaces/Core/Portable/Utilities/SerializableBytes.cs

```diff
@@ -244,6 +245,12 @@ public byte[] ToArray()
                 return array;
             }
 
+            public ImmutableArray<byte> ToImmutableArray()
+            {
+                var array = ToArray();
+                return ImmutableArrayExtensions.DangerousCreateFromUnderlyingArray(ref array);
```

Comment:
Do we really need to do this? Can't we do the same thing by creating a builder and filling it?

At least with the Span API we're eventually going to remove it, but I don't think there is an equivalent API on ImmutableArray, is there?

Reply:
I didn't want to rewrite to much of the SerializableBytes implementation. Feel free to make it better. I think having the `DangerousCreateFromUnderlyingArray` API is good - it is useful every now and then.

Reply:
I don't think convenience is a good enough reason to add an API this dangerous.

Reply:
We already have the dangeous API here that goes the other direction. I don't see the point of having the other one elsewhere.

Reply:
That method has a proposal to remove it when new APIs are available from ImmutableArray. I don't see a proposal to remove this API ever.

Reply:
I'll file an issue if that helps.

Reply:
An issue to do what? Remove it when the API is available on ImmutableArray itself?

Reply:
Issue to update SerializableBytes to not use the API.

Reply:
I don't think we should file issues to fix mistakes that we can easily not make in the first place.

Reply:
I have to draw a line somewhere. I can't update all code to use immutable arrays. I have other things to do.

Reply:
Looked into this again and I believe using `DangerousCreateFromUnderlyingArray` is actually the correct approach here, not just a workaround. The `SerializableBytes` type implements a custom `Stream`. `Stream` APIs work with `byte[]`. We could duplicate the implementation of the [helper](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Utilities/SerializableBytes.cs,212) that's populating they `byte[]` to work with ImmutableArray builder but I don't think it's desirable to have a copy of the code. The sole purpose of `DangerousCreateFromUnderlyingArray` is to be a bridge between mutable and immutable array APIs, which is exactly how it's used here.